### PR TITLE
Removed docker-compose.override.yml file and updated tests

### PR DESCRIPTION
**Pull request by m1yag1**
* Updated test database config to include docker defaults.
* Updated postgres database image from 10.5 to 11.
* Deleted docker-compose.override.yml file.
* Upgraded docker-compose.yml version to 3.5 and used gunicorn run
command with --reload.
* Added make test command to run tests in docker-compose
* Refactored make_database_url test into two tests.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,4 +1,21 @@
...
 services:
   web:
-    build: ./
+    build:
+      context: .
+      dockerfile: Dockerfile
+    volumes:
+      - .:/app
+    ports:
+      - "5000:5000"
```
Comment (karenc) [inline docker-compose.yml]: Is it possible to use `FLASK_PORT` here too?

Comment (m1yag1) [reply docker-compose.yml]: I thought I did that... https://github.com/openstax/qualitas-server/commit/d0a7afbbb6d59a73e009031a3c4306d4d6888274

File: travis-compose.yml [hunk 2]
```diff
@@ -16,6 +18,6 @@ services:
     links:
       - db
   db:
-    image: postgres:10.5-alpine
+    image: postgres:11-alpine
```
Comment (karenc) [inline travis-compose.yml]: Is this going to be a problem when we deploy to production / heroku?  I'm getting this when I do `docker-compose up`:

```
db_1   | 2019-05-10 00:49:38.112 UTC [1] DETAIL:  The data directory was initialized by PostgreSQL version 10, which is not compatible with this version 11.2.
qualitasserver_db_1 exited with code 1
```

File: docker-compose.yml (hunk 1 above)
Comment (m1yag1) [reply docker-compose.yml]: Oh I see! I was on phone earlier.

File: travis-compose.yml (hunk 2 above)
Comment (m1yag1) [reply travis-compose.yml]: yea... you'll need to do `docker-compose up -V` this will recreate the volumes.

**Pull request merged**